runtime/src/main/java/org/corfudb/protocols/wireprotocol/ExceptionMsg.java

```diff
@@ -0,0 +1,78 @@
+package org.corfudb.protocols.wireprotocol;
+
+import io.netty.buffer.ByteBuf;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+
+import org.corfudb.util.serializer.Serializers;
+
+/** This message wraps exceptions that are encountered on the server.
+ *
+ */
+public class ExceptionMsg implements ICorfuPayload<ExceptionMsg> {
+
+    /** The throwable that was thrown remotely. */
+    @Getter
+    final Throwable throwable;
+
+    /** Constructor for ExceptionMsg.
+     *
+     * @param t     The throwable to send.
+     */
+    public ExceptionMsg(Throwable t) {
+        throwable = t;
+    }
+
+    /** Represents an exception which failed serialization. */
+    public static class SerializationFailedException extends Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename this class to remove "Exception" or correct its inheritance. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2166)